base/game_events/GameEventManager.js

```diff
@@ -59,9 +60,10 @@ export class GameEventManager {
                 this.fire_npc_events(npc);
             }
         } else if (npc.npc_type === npc_types.SHOP) {
-            //open the shop here
+            if(!this.data.shop_open){
+                this.data.shop_screen.open_menu(npc.shop_key);
+            }
 
-            //on shop finish:
             this.on_event = false;
             this.data.force_stop_movement = false;
             this.control_enable = true;
```

Comment:
These 3 lines should be called when the shop is finished (on shop finish callback).